src/components/PopupWithImage.js

```diff
@@ -7,9 +7,9 @@ export default class PopUpWithImage extends Popup {
 
     super(popupSelector);
 
-    /* this._name = name;
+    this._name = name;
 
-    this._link = link; */
+    this._link = link;
 
     this._popupElement = document.querySelector(popupSelector);
```

Comment:
`    this._popupElement = document.querySelector(popupSelector);
`

**No** need to find the popup again in the inheritors of class `Popup` because the parent has already found it in its constructor and it’s `this._popupElement`. It’s available in any inheritor.

Reply:
It looks like when I delete this line, my cards are unable to be created because the other variables become undefined. Do i need to add this to the constructor and super selector?

Reply:
> It looks like when I delete this line, my cards are unable to be created because the other variables become undefined. Do i need to add this to the constructor and super selector?

There is a mistake in line 5 

It should be 


```js

    super({ popupSelector });

```